### ISSUE TITLE
TvApp

### ISSUE THREAD

**Issue by skiely** [interlark/ustvgo_downloader#65]
Hello, just wondering if possible to make new app with https://thetvapp.to/ ?
Thanks

**Comment by simvande** [comment]
Yes, https://thetvapp.to/  is reliable. Just it needs the token generation.

**Comment by Nostradamus1973** [comment]
Good morning @interlark 
I wanted to ask you for some help on another project. I know you're busy, however, I figured this might be something you'd have interest in.
Were you able to get a working script to run with the [https://thetvapp.to](https://thetvapp.to) site?

This topic recently came up over in the [Cabernet Git](https://github.com/cabernetwork/cabernet/discussions/112#discussioncomment-8405395) recently and I figured that maybe you could lend your expertise to help another cause.

You may even find some similarities to your interests there, the guy running the git over there is really quite talented and he has a few people under him that really like helping other people to get Cabernet working for regular guys like myself.

If not, I totally understand.

Have a great day.  :)

**Comment by interlark** [comment]
@Nostradamus1973 Hi!

I think I can get it working, if somebody prepares the TV-channel icons for the EPG repo :) See examples [here](https://github.com/interlark/ustvgo-tvguide/tree/master/images/icons) or [here](https://github.com/interlark/123tv-tvguide/tree/master/images/icons).

**Comment by Nostradamus1973** [comment]
Good morning Andy,
I sent a private message via discord to the creator of Cabernet a few days back, I never heard back from him, so, I'm thinking he might be busy and I'd rather not get too pushy with him since he's always been good to me.

So, instead, maybe we can get something working and then see if it's something he might be interested in incorporating as a plugin down the road.
I can definitely grab the icons. I'll grab, dark and light versions that are transparent and larger than 300x300pix. Sound good?
Thx Andy :)

**Comment by interlark** [comment]
@Nostradamus1973
Ok, just make sute all the icons are square sized, like 300x300, 512x512. 
Also, you can grab light icons for dark themes [here](https://github.com/tv-logo/tv-logos).
I'll code the rest things, just like [ustvgo-iptv](https://github.com/interlark/ustvgo-iptv) / [ustvgo-tvguide](https://github.com/interlark/ustvgo-tvguide).

I checked out [thetvapp](https://thetvapp.to/) and kinda liked it, despite that it only works with VPN on.

**Comment by Nostradamus1973** [comment]
Good morning @interlark 

Okay Andy, I'll look for square ones that aren't on your list you provided, since we can obviously reuse those icons that you already have.
Also, I'll flag @rocky4546 that way if he wants to follow along or give us direction, that would be helpful. I have no idea if he has any interest in incorporating your work into his script, if he doesn't we'll continue on as planned. I say this because he has a wicked nice TV Guide Plugin that may already support what we're doing here; however,  I hardly know anything about coding to make that assumption.

Give me a few days to work on those icons, we have a big storm hitting Rhode Island and I'll be knackered after all the snow removal.

Thanks and have a nice day.

**Comment by habitual69** [comment]
I've been developing a script for scraping stream links from thetvapp.to, and I'm excited to share it with the community: [TheTvApp StreamScraper](https://github.com/habitual69/thetvappstream). I warmly invite everyone to take a look and explore its features. Your feedback is incredibly valuable to me, so please don't hesitate to share any suggestions or ideas for enhancements. Thank you for your time and for considering contributing to the project's improvement!

**Comment by ttx99** [comment]
@interlark  Please create a CLI that plays the stream in mpv player.

**Comment by csrwhite** [comment]
@interlark
Good day Sir,
Just asking if there has been any progress on this. I really am looking forward to the result of the script. 
And Kudos to you and your excellent work.

**Comment by Nostradamus1973** [comment]
@csrwhite I'm terribly sorry I've been out of it for a while, I've been experiencing some health issues that are almost over, I hope.
You may want to check out @habitual69 script for the time being.
Have a nice day.

**Comment by skiely** [comment]
> @csrwhite I'm terribly sorry I've been out of it for a while, I've been experiencing some health issues that are almost over, I hope.
> You may want to check out @habitual69 script for the time being.
> Have a nice day.

@habitual69 script doesn't work anymore.

**Comment by csrwhite** [comment]
@Nostradamus1973 
Hey thanks for the heads up. Sorry to hear about the health issues but good to know you are getting over it. I was about to echo @skiely  note. I did check out the script from @habitual69 but indeed it has some issues as it doesn't work to write an m3u8 playlist like Andy's script.

I will be patient while you guys do your thing. you are the real MVPs here.

Thanks again.
